components/base/wheeled/sensorbase_test.go

```diff
@@ -317,3 +321,138 @@ func TestSensorBase(t *testing.T) {
+				return &movementsensor.Properties{
+					OrientationSupported:     true,
+					AngularVelocitySupported: true,
+					LinearVelocitySupported:  true,
+				}, errors.New("bad sesnor")
+			}
+			deps[movementsensor.Named(msName)] = ms
+		default:
+		}
+	}
+	return deps, cfg
+}
+
+func TestReconfig(t *testing.T) {
+	ctx := context.Background()
+	logger := golog.NewTestLogger(t)
+	lmNames := []string{"l-m"}
+	rmNames := []string{"r-m"}
+
+	deps, cfg := msDependencies(t, lmNames, rmNames, []string{"orientation"}, logger)
+
+	b, err := createWheeledBase(ctx, deps, cfg, logger)
+	test.That(t, err, test.ShouldBeNil)
+
+	deps, cfg = msDependencies(t, lmNames, rmNames, []string{"orientation1"}, logger)
+	err = b.Reconfigure(ctx, deps, cfg)
+	test.That(t, err, test.ShouldBeNil)
+	sb, ok := b.(*sensorBase)
+	test.That(t, ok, test.ShouldBeTrue)
+	test.That(t, sb.orientation.Name().ShortName(), test.ShouldResemble, "orientation1")
+
+	deps, cfg = msDependencies(t, lmNames, rmNames, []string{"orientation2"}, logger)
+	err = b.Reconfigure(ctx, deps, cfg)
+	test.That(t, err, test.ShouldBeNil)
+	sb, ok = b.(*sensorBase)
+	test.That(t, ok, test.ShouldBeTrue)
+	test.That(t, sb.orientation.Name().ShortName(), test.ShouldResemble, "orientation2")
+
+	deps, cfg = msDependencies(t, lmNames, rmNames, []string{"setvel1"}, logger)
+	err = b.Reconfigure(ctx, deps, cfg)
+	test.That(t, err, test.ShouldBeNil)
+	sb, ok = b.(*sensorBase)
+	test.That(t, ok, test.ShouldBeTrue)
+	test.That(t, sb.velocitiesSensor.Name().ShortName(), test.ShouldResemble, "setvel1")
+
+	deps, cfg = msDependencies(t, lmNames, rmNames, []string{"setvel2"}, logger)
+	err = b.Reconfigure(ctx, deps, cfg)
+	test.That(t, err, test.ShouldBeNil)
+	sb, ok = b.(*sensorBase)
+	test.That(t, ok, test.ShouldBeTrue)
+	test.That(t, sb.velocitiesSensor.Name().ShortName(), test.ShouldResemble, "setvel2")
+
+	deps, cfg = msDependencies(t, lmNames, rmNames, []string{"orientation3", "setvel3", "Bad"}, logger)
+	err = b.Reconfigure(ctx, deps, cfg)
+	test.That(t, err, test.ShouldBeNil)
+	sb, ok = b.(*sensorBase)
+	test.That(t, ok, test.ShouldBeTrue)
+	test.That(t, sb.orientation.Name().ShortName(), test.ShouldResemble, "orientation3")
+	test.That(t, sb.velocitiesSensor.Name().ShortName(), test.ShouldResemble, "setvel3")
+
+	deps, cfg = msDependencies(t, lmNames, rmNames, []string{"Bad", "orientation4", "setvel4"}, logger)
+	err = b.Reconfigure(ctx, deps, cfg)
+	test.That(t, err, test.ShouldBeNil)
+	sb, ok = b.(*sensorBase)
+	test.That(t, ok, test.ShouldBeTrue)
+	test.That(t, sb.orientation.Name().ShortName(), test.ShouldResemble, "orientation4")
+	test.That(t, sb.velocitiesSensor.Name().ShortName(), test.ShouldResemble, "setvel4")
+
+	deps, cfg = msDependencies(t, lmNames, rmNames, []string{"Bad"}, logger)
+	err = b.Reconfigure(ctx, deps, cfg)
+	test.That(t, sb.orientation, test.ShouldBeNil)
+	test.That(t, sb.orientation, test.ShouldBeNil)
```

Comment:
these two lines are duplicate, think the second is supposed to be `sb.velocitesSensor`